lib/specialist_tag_finder.rb

```diff
@@ -1,32 +1,33 @@
 class SpecialistTagFinder
 
-  def initialize(document)
-    @document = document
+  def initialize(edition)
+    @edition = edition
   end
 
-  def primary_sector_tag
-    primary_subsector_tag.parent if primary_subsector_tag
-  end
-
-  def primary_subsector_tag
-    if primary_tag_slug = @document.primary_specialist_sector_tag
-      specialist_sector_tags.find {|t| t.slug == primary_tag_slug }
-    end
-  end
+  def topics
+    presented_edition = PublishingApiPresenters::Edition.new(@edition)
+    edition_path = presented_edition.base_path
+    content_item = Whitehall.content_store.content_item(edition_path)
 
-  def sectors_and_subsectors
-    specialist_sector_tags.map { |t| [t, t.parent] }.flatten.compact.uniq
+    return [] unless content_item
+    Array(content_item.links["topics"])
   end
 
-private
-
-  def artefact
-    @artefact ||= Whitehall.content_api.artefact(RegisterableEdition.new(@document).slug)
+  def grandparent_topic
```

Comment:
It might be good to add a comment explaining what the grandparent topic is, or perhaps rename it to clarify that it is the top level topic the edition falls under. I'm not sure I understand it correctly. I'm assuming:

* The parent link is always a topic
* The topic hierarchy is 2 levels deep
* We can only tag to the bottom level
* We only display the top level in the breadcrumb (previously known as the primary specialist sector)

E.g. an edition parent would be environmental-management/chemicals and then the grandparent is environmental-management

Is this correct?